### PR TITLE
Feature/fixed price

### PR DESCRIPTION
**Pull request by mihaisc**
Changed publish so it created a pool or adds fixed price. Publish looks like this now
`const priceOptions = { price: 10, tokensToMint: 10, type: 'simple', weightOnDataToken: '', liquidityProviderFee: '' } `
`const ddo = await publish(asset as Metadata, priceOptions, 'access', '', '')`

`price` is only used for `simple` while `weightOnDataToken` and `liquidityProviderFee` is used only for `advanced`

I think this should be merged after #104

### CODE REVIEW

File: src/hooks/usePublish/PriceOptions.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export interface PriceOptions {
+  price?: number
+  tokensToMint: number
+  type: 'simple' | 'advanced' | string
```
Comment (kremalicious) [inline src/hooks/usePublish/PriceOptions.ts:4]: let's unify the naming, discussed somewhere on Slack. So we should now name pricing options everywhere:

- `fixed`
- `dynamic`

Comment (mihaisc) [reply src/hooks/usePublish/PriceOptions.ts]: done

**Comment by kremalicious** [comment]
And I was seeing this temp hack being removed in here somewhere, but think latest merge leaves that again https://github.com/oceanprotocol/react/blob/main/example/src/NetworkMonitor.tsx#L12

If we can now get network based on chainId then we can remove the hack indeed

**Comment by mihaisc** [comment]
hmmm, actually fixed that, but I think that I was to quick with resolving conflicts and removed it ...

File: example/src/Publish.tsx [hunk 2]
```diff
@@ -32,14 +31,19 @@ export function Publish() {
...
-    //   '90',
-    //   '9',
-    //   '0.03'
-    // )
+    console.log(isLoading)
+
+    const priceOptions = {
+      price: 10,
+      tokensToMint: 10,
+      type: 'simple',
```
Comment (kremalicious) [inline example/src/Publish.tsx:39]:
```suggestion
      type: 'fixed',
```

File: src/hooks/usePublish/usePublish.ts [hunk 3]
```diff
@@ -40,23 +40,23 @@ function usePublish(): UsePublish {
   /**
    * Publish an asset.It also creates the datatoken, mints tokens and gives the market allowance
    * @param  {Metadata} asset The metadata of the asset.
-   * @param  {string} tokensToMint Numer of tokens to mint and give allowance to market
-   * @param  {ServiceConfig[]} serviceConfigs Desired services of the asset, ex: [{serviceType: 'access', cost:'1'}]
+   * @param  {PriceOptions}  priceOptions : number of tokens to mint, datatoken weight , liquidity fee, type : simple, advanced
```
Comment (kremalicious) [inline src/hooks/usePublish/usePublish.ts:43]:
```suggestion
   * @param  {PriceOptions}  priceOptions : number of tokens to mint, datatoken weight , liquidity fee, type : fixed, dynamic
```

**Review by kremalicious — APPROVED** [review]

**Pull request merged**